ebi_eva_common_pyutils/assembly/assembly.py

```diff
@@ -29,3 +31,39 @@ def get_supported_asm_from_ensembl(tax_id: int) -> str:
     if assembly_accession_attribute in response:
         return str(response.get(assembly_accession_attribute))
     return None
+
+
+@lru_cache
+def get_taxonomy_to_assembly_mapping_from_ensembl_rapid_release():
+    """
+    Returns a dict mapping taxonomy ID to assembly accession, choosing the most recently released,
+    lexicographically first, non-alternate haplotype assembly when multiple are present.
+    """
+    list_data = json_request('https://ftp.ensembl.org/pub/rapid-release/species_metadata.json')
+    results = {}
+    for asm_data in list_data:
+        tax_id = asm_data['taxonomy_id']
+        asm_accession = asm_data['assembly_accession']
+        strain = asm_data['strain']
+        release_date = datetime.strptime(asm_data['release_date'], '%Y-%m-%d')
+
+        # If we haven't seen this taxonomy before, just use this assembly
+        if tax_id not in results:
+            results[tax_id] = (asm_accession, release_date)
+            continue
+
+        # Skip alternate haplotype assemblies
+        if strain and strain.lower() == 'alternate haplotype':
+            continue
+        current_assembly, current_date = results[tax_id]
+        # Keep the more recent assembly, or the lexicographically first one if release dates are equal
+        if current_date < release_date or (current_date == release_date and asm_accession < current_assembly):
+            results[tax_id] = (asm_accession, release_date)
+
+    return {key: val[0] for key, val in results.items()}
```

Comment:
**Optional:** I am wondering if there is value in storing the release dates somewhere as well for our reference...